src/components/MessageInput/InputButtons.tsx

```diff
@@ -0,0 +1,215 @@
+
+import type {
+  DefaultAttachmentType,
+  DefaultChannelType,
+  DefaultCommandType,
+  DefaultEventType,
+  DefaultMessageType,
+  DefaultReactionType,
+  DefaultUserType,
+  UnknownType,
+} from '../../types/types';
+
+const styles = StyleSheet.create({
+  attachButtonContainer: { paddingRight: 10 },
+});
+
+export type InputButtonsProps<
+  At extends DefaultAttachmentType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+> = Partial<InputButtonsWithContextProps<At, Ch, Co, Ev, Me, Re, Us>>;
+
+export type InputButtonsWithContextProps<
+  At extends DefaultAttachmentType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+> = Pick<
+  MessageInputContextValue<At, Ch, Co, Ev, Me, Re, Us>,
+  | 'AttachButton'
+  | 'CommandsButton'
+  | 'giphyActive'
+  | 'hasCommands'
+  | 'hasFilePicker'
+  | 'hasImagePicker'
+  | 'MoreOptionsButton'
+  | 'setShowMoreOptions'
+  | 'showMoreOptions'
+  | 'text'
+  | 'uploadsEnabled'
+> & {
+  closeAttachmentPicker?: () => void;
+  openAttachmentPicker?: () => void;
+  openCommandsPicker?: () => void;
+  toggleAttachmentPicker?: () => void;
+};
+
+export const InputButtonsWithContext = <
+  At extends DefaultAttachmentType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+>(
+  props: InputButtonsWithContextProps<At, Ch, Co, Ev, Me, Re, Us>,
+) => {
+  const {
+    AttachButton,
+    CommandsButton,
+    giphyActive,
+    hasCommands,
+    hasFilePicker,
+    hasImagePicker,
+    MoreOptionsButton,
+    openCommandsPicker,
+    setShowMoreOptions,
+    showMoreOptions,
+    text,
+    toggleAttachmentPicker,
+    uploadsEnabled,
+  } = props;
+
+  const {
+    theme: {
+      messageInput: { attachButtonContainer, commandsButtonContainer },
+    },
+  } = useTheme();
+
+  if (giphyActive) {
+    return null;
+  }
+
+  return (
+    <>
+      {!showMoreOptions && (hasImagePicker || hasFilePicker) && hasCommands ? (
+        <MoreOptionsButton handleOnPress={() => setShowMoreOptions(true)} />
+      ) : (
+        <>
+          {(hasImagePicker || hasFilePicker) && uploadsEnabled !== false && (
+            <View
+              style={[
+                hasCommands ? styles.attachButtonContainer : undefined,
```

Comment:
undefined work fine here? or does it need to be
```suggestion
                hasCommands ? styles.attachButtonContainer : {},
```

Reply:
Yup, undefined works as well!!